src/read.jl

```diff
@@ -39,33 +41,25 @@ to `identity`.
 edf_unknown(f, field::AbstractString) = field == "X" ? missing : f(field)
 edf_unknown(field::AbstractString) = edf_unknown(identity, field)
 
-"""
-    set!(io::IO, [f=strip,] sigs::Vector{Signal}, name::Symbol, sz::Integer)
+#####
+##### Reading EDF Files
+#####
 
-For each signal in `sigs`, set the field given by `name` to the result of `f` applied
-to a chunk of size `sz` read from `io` as a `String`.
-"""
-function set!(io::IO, f, sigs::Vector{Signal}, name::Symbol, sz::Integer)
-    n = length(sigs)
-    T = fieldtype(Signal, name)
-    @inbounds for i = 1:n
-        x = f(String(Base.read(io, sz)))
-        setfield!(sigs[i], name, convert(T, x))
-    end
-    return sigs
-end
 
-function set!(io::IO, sigs::Vector{Signal}, name::Symbol, sz::Integer)
-    return set!(io, strip, sigs, name, sz)
+function read_file_and_signal_headers(io::IO)
+    file_header, header_byte_count, signal_count = read_file_header(io)
+    fields = [method(String(Base.read(io, size)))
+              for signal in 1:signal_count, (size, method) in zip(FIELD_SIZES, PARSE_METHODS)]
+    T = Union{SignalHeader,AnnotationListHeader}
+    signal_headers = T[SignalHeader(row...) for row in eachrow(fields)]
```

Comment:
```suggestion
    signal_headers = T[SignalHeader(fields[i, :]...) for i in 1:size(fields, 1)]
```
should work with 1.0